### PR TITLE
Check for offshore and test

### PR DESCRIPTION
**Pull request by ximenesuk**
This PR adds the additional check for whether a location is completely outside both the UK EEA Offshore area and Northern Ireland landmass. Locations outside of this area will currently report two error messages:
```
{'desc': 'NATE / NATN outside UK Offshore EEA or Northern Ireland (Paris)', 'group': 'LOCA', 'line': '16'},
{'desc': 'NATE / NATN outside Great Britain or Northern Ireland landmass (Paris)', 'group': 'LOCA', 'line': '16'},
```
It would be possible to drop this to the single "outside UK Offshore EEA" error if that was preferred.

Locations within the offshore area but not on the UK landmass will report a single error:
```
{'desc': 'NATE / NATN outside Great Britain or Northern Ireland landmass (MorayFirth)', 'group': 'LOCA', 'line': '15'},
```
The unit tests should pass.

### CODE REVIEW

File: app/bgs_rules.py [hunk 1]
```diff
@@ -191,15 +194,23 @@ def check_loca_within_great_britain(tables: dict) -> List[dict]:
...
     inside_gb_mask = location.intersects(gb_outline)
     as_irish_grid = location.to_crs("EPSG:29903")
     inside_ni_mask = as_irish_grid.intersects(ni_outline)
+    outside_uk_eea_and_ni_mask = ~inside_uk_eea_mask & ~inside_ni_mask
     outside_gb_and_ni_mask = ~inside_gb_mask & ~inside_ni_mask
 
+    for loca_id, row in location.loc[outside_uk_eea_and_ni_mask].iterrows():
+        errors.append({
+            'line': f'{row["line_no"]}', 'group': 'LOCA',
+            'desc': f'NATE / NATN outside UK Offshore EEA or Northern Ireland ({loca_id})'
```
Comment (KoalaGeo) [inline app/bgs_rules.py:207]: Change to "f'NATE / NATN outside UK Offshore EEA or Onshore Northern Ireland."

File: app/bgs_rules.py [hunk 2]
```diff
@@ -191,15 +194,23 @@ def check_loca_within_great_britain(tables: dict) -> List[dict]:
...
+        })
+
     for loca_id, row in location.loc[outside_gb_and_ni_mask].iterrows():
         errors.append({
             'line': f'{row["line_no"]}', 'group': 'LOCA',
-            'desc': f'NATE / NATN outside Great Britain and Northern Ireland ({loca_id})'
+            'desc': f'NATE / NATN outside Great Britain or Northern Ireland landmass ({loca_id})'
```
Comment (KoalaGeo) [inline app/bgs_rules.py:213]: Ammend to "outside Onshore Great Britain or Northern Ireland boundaries"

File: test/unit/test_bgs_rules.py [hunk 3]
```diff
@@ -122,20 +122,33 @@ def test_drill_depth_geol_record():
...
-         'group': 'LOCA',
-         'line': '3'},
-        {'desc': 'NATE / NATN outside Great Britain and Northern Ireland (Derry)',  # these coords given in EPSG:27700
-         'group': 'LOCA',
-         'line': '4'},
-        {'desc': 'NATE / NATN in Northern Ireland but LOCA_GREF undefined (Belfast)',
-         'group': 'LOCA',
-         'line': '6'}
-    ]
+    expected = [{'desc': 'NATE / NATN outside UK Offshore EEA or Northern Ireland (Bad NATE)',
```
Comment (KoalaGeo) [inline test/unit/test_bgs_rules.py:125]: Ammend to match the defined error

**Comment by KoalaGeo** [comment]
@ximenesuk looks good & works for me. 

If we can slightly ammend the error message think it's good to merge

**Comment by ximenesuk** [comment]
> @ximenesuk looks good & works for me.
> 
> If we can slightly ammend the error message think it's good to merge

Messages amended.

**Comment by KoalaGeo** [comment]
Addresses https://github.com/BritishGeologicalSurvey/pyagsapi/issues/102

**Pull request merged**